src/main/scala/com.snowplowanalytics.iglu/client/Resolver.scala

```diff
@@ -295,26 +288,26 @@ case class Resolver(
 
   private[client] val allRepos = Bootstrap.Repo :: repos
 
-  /**
-   * Our LRU cache.
-   */
+  // TODO: ideally replace unsafeRunSync() with referentially transparent API
   object cache {
 
     private val lru: Option[SchemaLruMap] =
-      if (cacheSize > 0) Some(new SchemaLruMap(cacheSize)) else None
+      if (cacheSize > 0)
+        Some(LruMap.create[IO, SchemaKey, SchemaLookupStamped](cacheSize).unsafeRunSync())
```

Comment:
We either keep the non-RT API in 0.6 with `unsafeRunSync()` calls in cache or we need to refactor the whole API to be referentially transparent.

Reply:
I vote for fully-RT API. Mixing eager and strict evaluations is just an another huge source of bugs as we saw in Scala Tracker.